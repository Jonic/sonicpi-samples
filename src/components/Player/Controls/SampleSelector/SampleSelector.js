import PropTypes from 'prop-types'
import React from 'react'

import './SampleSelector.css'

const SampleSelector = ({
  currentSample,
  filteredSamples,
  isLoading,
  sampleSelectorChangeHandler,
}) => {
  const handleSampleChange = (event) => {
    sampleSelectorChangeHandler(event.target.value)
  }

  return (
    <div className="c-SampleSelector">
      Sample Selector:

      <select
        className="c-SampleSelector__dropdown"
        disabled={isLoading}
        value={currentSample}
        onChange={handleSampleChange}
      >
        {
          filteredSamples.map((sample, index) => {
            return (
              <option
                key={index}
                value={sample}
              >
                {sample}
              </option>
            )
          })
        }
      </select>
    </div>
  )
}

SampleSelector.propTypes = {
  currentSample:               PropTypes.string.isRequired,
  filteredSamples:             PropTypes.array.isRequired,
  isLoading:                   PropTypes.bool.isRequired,
  sampleSelectorChangeHandler: PropTypes.func.isRequired,
}

export default SampleSelector
