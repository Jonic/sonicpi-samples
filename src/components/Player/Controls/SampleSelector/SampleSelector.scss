@import '../../../../styles/component-helper';

.c-sample-selector {}
.c-sample-selector__select {}
