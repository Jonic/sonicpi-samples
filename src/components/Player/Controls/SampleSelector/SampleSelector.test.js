import React from 'react'
import { shallow } from 'enzyme'
import SampleSelector from './SampleSelector'

const currentSample = ''
const filteredSamples = []
const isLoading = false
const sampleSelectorChangeHandler = jest.fn()

let wrapper

describe('<SampleSelector />', () => {
  beforeEach(() => {
    wrapper = shallow(
      <SampleSelector
        currentSample={currentSample}
        filteredSamples={filteredSamples}
        isLoading={isLoading}
        sampleSelectorChangeHandler={sampleSelectorChangeHandler}
      />
    )
  })

  describe('Renders...', () => {
    it('without crashing', () => {
      expect(wrapper.find('.c-SampleSelector')).toBePresent()
    })

    it('matches its snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
