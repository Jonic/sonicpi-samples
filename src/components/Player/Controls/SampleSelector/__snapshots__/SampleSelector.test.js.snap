// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleSelector /> Renders... matches its snapshot 1`] = `
<div
  className="c-sample-selector"
>
  Sample Selector:
   
  <Select
    className="c-sample-selector__select"
    disabled={false}
    onChange={[Function]}
    options={Array []}
    value=""
  />
</div>
`;
